mongo/find.go

```diff
@@ -73,6 +77,13 @@ type (
 // Find executes a find mongo query by using the provided FindParams, fills the passed in result
 // slice pointer and returns a Cursor.
 func Find(p FindParams, results interface{}) (Cursor, error) {
+	ctx := context.Background()
+	return FindWithContext(ctx, p, results)
+}
+
+// FindWithContext executes a find mongo query by using the provided FindParams, fills the passed in result
+// slice pointer and returns a Cursor.
+func FindWithContext(ctx context.Context, p FindParams, results interface{}) (Cursor, error) {
```

Comment:
The reason there was no context support to begin with is because mgo does not support it. I think we should simply make this a breaking changes to the Find function as it makes no sense to me that someone would not want to pass in context if it’s supported.